packages/react-native-bpk-component-button/src/BpkButton.ios.js

```diff
@@ -20,100 +20,239 @@
+      ? {
+          gradientStartColor: themeAttributes.buttonPrimaryGradientStartColor,
+          gradientEndColor: themeAttributes.buttonPrimaryGradientEndColor,
+        }
+      : {
+          gradientStartColor: gradientColors.primary.start,
+          gradientEndColor: gradientColors.primary.end,
+        };
+  }
+  if (type === BUTTON_TYPES.featured) {
+    if (disabled) {
+      return {
+        gradientStartColor: backgroundColors.disabled,
+        gradientEndColor: backgroundColors.disabled,
+      };
+    }
+    return themeAttributes
+      ? {
+          gradientStartColor: themeAttributes.buttonFeaturedGradientStartColor,
+          gradientEndColor: themeAttributes.buttonFeaturedGradientEndColor,
+        }
+      : {
+          gradientStartColor: gradientColors.featured.start,
+          gradientEndColor: gradientColors.featured.end,
+        };
+  }
+  if (type === BUTTON_TYPES.secondary) {
+    if (disabled) {
+      return {
+        backgroundColor: backgroundColors.disabled,
+        borderColor: borderColors.disabled,
+      };
+    }
+    return themeAttributes
+      ? {
+          backgroundColor: themeAttributes.buttonSecondaryBackgroundColor,
+          borderColor: themeAttributes.buttonSecondaryBorderColor,
+        }
+      : {
+          backgroundColor: backgroundColors.secondary,
+          borderColor: borderColors.secondary,
+        };
+  }
+  if (disabled) {
+    return {
+      backgroundColor: backgroundColors.disabled,
+      borderColor: borderColors.disabled,
+    };
+  }
+  return themeAttributes
+    ? {
+        backgroundColor: themeAttributes.buttonDestructiveBackgroundColor,
+        borderColor: themeAttributes.buttonDestructiveBorderColor,
+      }
+    : {
+        backgroundColor: backgroundColors.destructive,
+        borderColor: borderColors.destructive,
+      };
+};
+
+const getTextColor = (
+  type: ButtonType,
+  themeAttributes: ?Theme,
+  disabled: boolean,
+) => {
+  if (disabled) {
+    return textColors.disabled;
+  }
+  switch (type) {
+    case BUTTON_TYPES.primary:
+      return themeAttributes
+        ? themeAttributes.buttonPrimaryTextColor
+        : textColors.primary;
+    case BUTTON_TYPES.featured:
+      return themeAttributes
+        ? themeAttributes.buttonFeaturedTextColor
+        : textColors.featured;
+    case BUTTON_TYPES.secondary:
+      return themeAttributes
+        ? themeAttributes.buttonSecondaryTextColor
+        : textColors.secondary;
+    default:
+      return themeAttributes
+        ? themeAttributes.buttonDestructiveTextColor
+        : textColors.destructive;
+  }
+};
+
 const BpkButton = (props: Props) => {
   const {
     accessibilityLabel,
     disabled,
     icon,
+    iconAlignment,
     iconOnly,
     large,
-    onPress,
+    theme,
     title,
     type,
-    theme: themeProp,
     ...rest
   } = props;
 
-  if (!Object.values(BUTTON_TYPES).includes(type)) {
+  if (!Object.keys(BUTTON_TYPES).includes(type)) {
```

Comment:
why the change? If users do `BUTTON_TYPES.primary` they are providing the value so I guess is better to check the value here, even though they are the same atm

Reply:
The keys and values are the same so I figured it would be more consistent with the Flow type and prop type if they matched:

```
export type ButtonType = $Keys<typeof BUTTON_TYPES>;

  type: PropTypes.oneOf(Object.keys(BUTTON_TYPES)),

```